### ISSUE TITLE
Benchmark the public API methods

### ISSUE THREAD

**Issue by gajus** [reactjs/redux#1338]
Specifically, the following methods:
- `createStore()`
  - `#getState()`
  - `#dispatch()`
  - `#subscribe()`
  - `#replaceReducer()`
- `bindActionCreators()`

I don't think `combineReducers()`, `applyMiddleware()` or `compose()` are worth benchmarking since these are called only during the setup phase and do not participate in the update cycle.

An example of a [benchmark written for `redux-immutable`](https://github.com/gajus/redux-immutable/blob/master/benchmarks/index.js).

**Comment by gajus** [comment]
I am happy to contribute the base benchmarks. Someone with more experience in benchmarking will be able to contribute/ guide towards making them more useful.

**Comment by gaearon** [comment]
I'm all for this. I'm not sure how to make them realistic though. Maybe we can copy some reducers from the sample apps and use them.

I'm also not sure how to avoid the noise in benchmarks. I tried Benchmark.js with very simple reducers and I was getting very large fluctuation in results on every run. Maybe it's because my reducers and subscribers were too simple.

On the other hand maybe we want something like https://github.com/facebook/react/tree/master/scripts/bench, with full control over what's happening in the bench. However I'm out of my depth here.

**Comment by gajus** [comment]
Me too. Will give it a couple of days to see if anyone suggests either an alternative to Benchmark.js or suggest a strategy for benchmarking before proceeding writing any tests.

**Comment by markerikson** [comment]
THREAD RESURRECTION!!!

@gajus , you still have any interest in trying to do this?  If not, we might as well close it.  (Y'know, a year and a half after it was opened.)

FWIW, I recently got into a bit of a discussion over in the repo for a Redux clone that claimed to have a faster approach to reducers, and wound up with a couple interesting bits of benchmarks: https://github.com/dogada/fast-redux/issues/1 .

Looking back at the original question: I don't think there's any value in benchmarking `getState` or `subscribe` - both of those should be trivial operations.  `dispatch` is by far the most relevant think to benchmark, and per that issue I linked, the dominant aspect is likely to be notifying the subscribers.
